dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageContextBuilder.java

```diff
@@ -0,0 +1,294 @@
+        ctxMap.put("VTLSERVLET_URI", UtilMethods.encodeURIComponent(pageIdent.getURI()));
+        ctxMap.put("HTMLPAGE_REDIRECT", UtilMethods.espaceForVelocity(htmlPage.getRedirect()));
+        ctxMap.put("pageTitle", UtilMethods.espaceForVelocity(htmlPage.getTitle()));
+        ctxMap.put("friendlyName", UtilMethods.espaceForVelocity(htmlPage.getFriendlyName()));
+        ctxMap.put("HTML_PAGE_LAST_MOD_DATE", UtilMethods.espaceForVelocity(htmlPage.getFriendlyName()));
+        ctxMap.put("HTMLPAGE_MOD_DATE", new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS").format(htmlPage.getModDate()));
+
+
+
+    }
+
+
+
+    private void populateContainers() throws DotDataException, DotSecurityException {
+        Table<String, String, Set<String>> pageContents = new MultiTreeAPI().getPageMultiTrees(htmlPage, mode.showLive);
+
+
+
+        if (!pageContents.isEmpty()) {
+            for (final String containerId : pageContents.rowKeySet()) {
+                for (final String uniqueId : pageContents.row(containerId)
+                    .keySet()) {
+                    Set<String> cons = pageContents.get(containerId, uniqueId);
+
+                    Container c = (mode.showLive) ? (Container) APILocator.getVersionableAPI()
+                        .findLiveVersion(containerId, user, false)
+                            : (Container) APILocator.getVersionableAPI()
+                                .findWorkingVersion(containerId, user, false);
+
+                    boolean hasWritePermissionOnContainer =
+                            permissionAPI.doesUserHavePermission(c, PERMISSION_WRITE, user, false) && APILocator.getPortletAPI()
+                                .hasContainerManagerRights(user);
+                    boolean hasReadPermissionOnContainer = permissionAPI.doesUserHavePermission(c, PERMISSION_READ, user, false);
+                    ctxMap.put("EDIT_CONTAINER_PERMISSION" + c.getIdentifier(), hasWritePermissionOnContainer);
+                    if (Config.getBooleanProperty("SIMPLE_PAGE_CONTENT_PERMISSIONING", true))
+                        ctxMap.put("USE_CONTAINER_PERMISSION" + c.getIdentifier(), true);
+                    else
+                        ctxMap.put("USE_CONTAINER_PERMISSION" + c.getIdentifier(), hasReadPermissionOnContainer);
+
+                    // to check user has permission to write this container
+                    boolean hasWritePermOverTheStructure = false;
+
+                    for (ContainerStructure cs : APILocator.getContainerAPI()
+                        .getContainerStructures(c)) {
+                        Structure st = CacheLocator.getContentTypeCache()
+                            .getStructureByInode(cs.getStructureId());
+
+                        hasWritePermOverTheStructure |= permissionAPI.doesUserHavePermission(st, PERMISSION_WRITE, user);
+                    }
+
+                    ctxMap.put("ADD_CONTENT_PERMISSION" + c.getIdentifier(), new Boolean(hasWritePermOverTheStructure));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "Boolean" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)